katdal/chunkstore_npy.py

```diff
@@ -93,6 +93,13 @@ def has_chunk(self, array_name, slices, dtype):
         filename = os.path.join(self.path, chunk_name) + '.npy'
         return os.path.exists(filename)
 
+    def list_chunk_ids(self, array_name):
+        """See the docstring of :meth:`ChunkStore.list_chunk_ids`."""
+        array_dir = os.path.join(self.path, array_name)
+        # Strip the .npy extension to get the chunk ID string
+        return [fn[:-4] for fn in os.listdir(array_dir)]
```

Comment:
This can go wrong if there happens to be other stuff in the directory apart from the .npy files. Maybe add `if fn.endswith('.npy')` to the list comprehension.

Reply:
OK, will add this test.

I was trying not to do too much (in the spirit of removing checks from has_chunk), since this list can also be pretty large (million elements?) and 99.999999% of the time that test will not be needed.